dice-where-downloader-lib/src/main/java/technology/dice/dicewhere/downloader/destination/s3/S3FileAcceptor.java

```diff
@@ -58,22 +63,29 @@ public StreamConsumer<Void> getStreamConsumer(
                   .storageClass(StorageClass.INTELLIGENT_TIERING)
                   .build();
           client.putObject(putObjectRequest, RequestBody.fromInputStream(stream, size));
-
           Latest latest = new Latest(clock.instant(), key);
           String latestContent = mapper.writeValueAsString(latest);
 
-          PutObjectRequest putLatest =
-              PutObjectRequest.builder()
-                  .key(Paths.get(key).getParent().toString() + "/" + LATEST_KEY)
-                  .bucket(bucket)
-                  .contentLength((long) latestContent.length())
-                  .storageClass(StorageClass.INTELLIGENT_TIERING)
-                  .build();
-          client.putObject(
-              putLatest,
-              RequestBody.fromInputStream(
-                  new StringInputStream(latestContent), latestContent.length()));
+          if ((!noMd5Check) && (!originalFileMd5.matches(stream.md5()))) {
```

Comment:
Same as above, can we move this check up so we avoid put + delete?

Reply:
No, because you only know the md5 of the transfer, *after* the transfer is complete